### DISCUSSION TITLE
No permission for remote acess when connecting to a local database via inet (isql)

### DISCUSSION THREAD

**Discussion by Prochy** [FirebirdSQL/firebird#8312] [Q&A]
I want to migrate from gsec to isql but I'm facing issue I cannot connect to the local fb when using following command:
`isql -x "127.0.0.1/3050:C:\Program Files\Firebird\security4.fdb" -u SYSDBA -p masterkey`

With gsec this works fine even with RemoteAccess = false.

Is this desired behavior? I understand the gsec uses different approach but I would suppose it should be possible to connect to the database from the loopbacks addresses even with isql.

Note: I cannot use xnet uri.

**Comment by mrotteveel** [comment]
Yes, this is the desired behaviour. Any connection over TCP/IP is considered remote access, even if it is to localhost. You need to use embedded or XNET.

However, if you want to manage users, you can connect to **any** database to manage users in the security database, as long as it doesn't use a custom security database. There is hardly ever a reason to connect to the security database itself.

As an aside, for future reference, please ask question on [firebird-support](https://groups.google.com/g/firebird-support) or on [GitHub discussions](https://github.com/FirebirdSQL/firebird/discussions). The issues are intended for reporting bugs or requesting new features/improvements.

**Reply by mrotteveel** [reply]
Minor qualification: you can only manage users in a database if it uses that security database *and* has the appropriate `UserManager` configured (either globally in `firebird.conf`, or specifically in `databases.conf`). A non-SYSDBA user must have the RDB$ADMIN role in the security database.

**Comment by mrotteveel** [comment]
NOTE: I have migrated this from Issues to Discussions
